File: fixtures/scale/kotlin/medium/src/calc/funcs/Rounding.kt
package calc.funcs

import calc.eval.FunctionRegistry
import kotlin.math.abs
import kotlin.math.ceil
import kotlin.math.floor
import kotlin.math.round
import kotlin.math.sign

fun registerRounding(registry: FunctionRegistry) {
    registry.addUnary("floor") { floor(it) }
    registry.addUnary("ceil") { ceil(it) }
    registry.addUnary("round") { round(it) }
    registry.addUnary("trunc") { it.toLong().toDouble() }
    registry.addUnary("abs") { abs(it) }
    registry.addUnary("sign") { sign(it) }
}

File: fixtures/scale/scala/medium/src/calc/funcs/Rounding.scala
package calc.funcs

import calc.eval.FunctionRegistry

object Rounding {
  def register(registry: FunctionRegistry): Unit = {
    registry.addUnary("floor", math.floor)
    registry.addUnary("ceil", math.ceil)
    registry.addUnary("round", x => math.round(x).toDouble)
    registry.addUnary("trunc", x => x.toLong.toDouble)
    registry.addUnary("abs", math.abs)
    registry.addUnary("sign", math.signum)
  }
}
